packages/@react-aria/tag/src/useTag.ts

```diff
@@ -10,74 +10,72 @@
  * governing permissions and limitations under the License.
  */
 
-import {ButtonHTMLAttributes, HTMLAttributes, KeyboardEvent, ReactNode} from 'react';
-import {DOMProps, Removable} from '@react-types/shared';
+import {ButtonHTMLAttributes, HTMLAttributes, KeyboardEvent} from 'react';
 import {filterDOMProps, mergeProps, useId} from '@react-aria/utils';
+import {GridState} from '@react-stately/grid';
 // @ts-ignore
 import intlMessages from '../intl/*.json';
+import {TagProps} from '@react-types/tag';
+import {useGridCell, useGridRow} from '@react-aria/grid';
 import {useMessageFormatter} from '@react-aria/i18n';
 
 
-export interface AriaTagProps extends Removable<ReactNode, void>, DOMProps {
-  children?: ReactNode,
-  isDisabled?: boolean,
-  validationState?: 'invalid' | 'valid',
-  isSelected?: boolean,
-  role?: string
-}
-
 export interface TagAria {
   tagProps: HTMLAttributes<HTMLElement>,
   labelProps: HTMLAttributes<HTMLElement>,
   clearButtonProps: ButtonHTMLAttributes<HTMLButtonElement>
 }
 
-export function useTag(props: AriaTagProps): TagAria {
+export function useTag(props: TagProps<any>, state: GridState<any, any>): TagAria {
   const {
     isDisabled,
-    validationState,
     isRemovable,
-    isSelected,
     onRemove,
     children,
-    role
+    item,
+    tagRef,
+    labelRef
   } = props;
   const formatMessage = useMessageFormatter(intlMessages);
   const removeString = formatMessage('remove');
   const tagId = useId();
   const buttonId = useId();
 
+  let {rowProps} = useGridRow({
+    node: item
+  }, state, tagRef);
+  let {gridCellProps} = useGridCell({
+    node: [...item.childNodes][0],
+    focusMode: 'cell'
+  }, state, labelRef);
+
   function onKeyDown(e: KeyboardEvent<HTMLElement>) {
     if (e.key === 'Delete' || e.key === 'Backspace') {
       onRemove(children, e);
       e.preventDefault();
     }
   }
   const pressProps = {
-    onPress: e => onRemove(children, e)
+    onPress: e => onRemove?.(children, e)
```

Comment:
IMO `onRemove` should probably be updated to accept a key (sourced from `item.key`) rather than children so that it matches our other event handlers and avoids the issue where two tags may have the same text. Open for feedback from the team